cnf-certification-test/amalpro/index.html

```diff
@@ -0,0 +1,247 @@
+      }
+
+      fieldset {
+        display: grid;
+        flex: 1 1 100%;
+      }
+
+      label {
+        display: block;
+        font-weight: var(--rh-font-weight-heading-bold, 700);
+        margin-block-end: var(--rh-space-sm, 6px);
+      }
+
+      rh-dialog::part(footer) {
+        padding-block-start: var(--rh-space-lg, 16px);
+      }
+
+      rh-footer-universal {
+        margin-block-start: auto;
+      }
+      main, :not(:defined) {
+        opacity: var(--main-opacity, 1);
+        transition: opacity 200ms ease;
+      }
+    </style>
+
+      <script type="module">
+      await Promise.all(['rh-button','pf-text-input'].map(x => customElements.whenDefined(x)));
+      document.body.style.removeProperty('--main-opacity');
+      </script>
+
+    <script type="module">
+      // import design system element definitions,
+      // which auto-register their tagnames once executed
+      import '@rhds/elements/rh-button/rh-button.js';
+      import '@rhds/elements/rh-dialog/rh-dialog.js';
+      import '@rhds/elements/rh-footer/rh-footer-universal.js';
+      import '@patternfly/elements/pf-text-input/pf-text-input.js';
+      // set up interactive elements
+      // use dynamic import to improve page load times
+      document.getElementById('logs').addEventListener('open', () => import('./logs.js'));
+      document.getElementById('form').addEventListener('submit', async function(event) {
+        // prevent page from redirecting to form endpoint.
+        // consider using a multi-page architecture instead.
+        event.preventDefault();
+        const { submit } = await import('./submit.js');
+        // submit the form
+        const { heading, message, state } = await submit(event.target);
+        // display the result
+        const { toast } = await import('./toast.js');
+        await toast({ heading, message, state, timeout: 20_000 });
+      });
+      document.getElementById('show-logs').addEventListener('click', function(event) {
+        const socket = new WebSocket('ws://localhost:8080/logstream');
+        const code = document
+        .getElementById('logs')
+        .querySelector('rh-code-block');
+        code.textContent = '';
+        // Handle incoming log messages
+        socket.addEventListener('message', function (event) {
+          code.textContent += event.data + '\n';
+        });
+      });
+    </script>
+
+  </head>
+  <body style="--main-opacity:0;">
+    <header>
+      <img alt="Red Hat"
+           src="https://static.redhat.com/libs/redhat/brand-assets/2/corp/logo--on-dark.svg"
+           width="100"
+           height="30">
+    </header>
+
+    <main>
+      <h1>CNF Certification Test</h1>
+
+      <form id="form">
+        <!-- see https://ux.redhat.com/patterns/form/ for design guidelines-->
+
+        <fieldset>
+          <legend>Environment Configuration</legend>
+          <label for="kubeConfigPath">Path of KubeConfig file:</label>
+          <pf-text-input id="kubeConfigPath"
```

Comment:
use a file picker to select kubeconfig

Reply:
you may add the `type="file"` attribute to `<pf-text-input>` for this

Reply:
See https://html.spec.whatwg.org/multipage/input.html#fakepath-srsly